src/firewall/core/rich.py

```diff
@@ -372,6 +368,19 @@ def __init__(self, family=None, rule_str=None, priority=0):
         if rule_str is not None:
             self._import_from_string(rule_str)
 
+        if priority is not None:
+            self.priority = priority
+
+        if family is not None:
+            family = str(family)
+            if self.family is None:
+                self.family = family
+            elif self.family != family:
+                raise FirewallError(errors.INVALID_FAMILY, family)
+
+        if rule_str is not None:
```

Comment:
Then a plain
```python
rule = firewall.core.rich.Rich_Rule()
```
would fail the check and raise an exception.

There are two modes of operation. One creating a rich-rule from a string (in which we call `check()`). The other is to create an empty rich-rule instance, and programmatically set the values. In the latter case, calling `check()` is pointless but also won't work, because the instance is not yet full initialized nor valid.

Reply:
ACK.